pluginframework/converters/go_to_tf.go

```diff
@@ -0,0 +1,224 @@
+		srcFieldName := field.StructField.Name
+
+		srcFieldTag := field.StructField.Tag.Get("json")
+		if srcFieldTag == "-" {
+			continue
+		}
+		destField := destVal.FieldByName(srcFieldName)
+
+		if !destField.IsValid() {
+			logger.Tracef(ctx, fmt.Sprintf("field skipped in gosdk to tfsdk conversion: destination struct does not have field %s", srcFieldName))
+			continue
+		}
+
+		err := goSdkToTfSdkSingleField(ctx, srcField, destField, fieldInForceSendFields(srcFieldName, forceSendFieldVal))
+		if err != nil {
+			return diag.Diagnostics{diag.NewErrorDiagnostic(err.Error(), "gosdk to tfsdk field conversion failure")}
+		}
+	}
+	return nil
+}
+
+func goSdkToTfSdkSingleField(ctx context.Context, srcField reflect.Value, destField reflect.Value, forceSendField bool) error {
+
+	if !destField.CanSet() {
+		panic(fmt.Errorf("destination field can not be set: %s. %s", destField.Type().Name(), common.TerraformBugErrorMessage))
+	}
+
+	srcFieldValue := srcField.Interface()
+
+	if srcFieldValue == nil {
+		return nil
+	}
+
+	switch srcField.Kind() {
+	case reflect.Ptr:
+		if srcField.IsNil() {
+			// Skip nils
+			return nil
+		}
+		destField.Set(reflect.New(destField.Type().Elem()))
+
+		// Recursively populate the nested struct.
+		if GoSdkToTfSdkStruct(ctx, srcFieldValue, destField.Interface()).HasError() {
+			panic(fmt.Sprintf("Error converting gosdk to tfsdk struct. %s", common.TerraformBugErrorMessage))
+		}
+	case reflect.Bool:
+		boolVal := srcFieldValue.(bool)
+		// check if the value is non-zero or if the field is in the forceSendFields list
+		if boolVal || forceSendField {
+			destField.Set(reflect.ValueOf(types.BoolValue(boolVal)))
+		} else {
+			destField.Set(reflect.ValueOf(types.BoolNull()))
+		}
+	case reflect.Int64:
+		// convert any kind of integer to int64
+		intVal := srcField.Convert(reflect.TypeOf(int64(0))).Int()
+		// check if the value is non-zero or if the field is in the forceSendFields list
+		if intVal != 0 || forceSendField {
+			destField.Set(reflect.ValueOf(types.Int64Value(intVal)))
+		} else {
+			destField.Set(reflect.ValueOf(types.Int64Null()))
+		}
+	case reflect.Float64:
+		// convert any kind of float to float64
+		float64Val := srcField.Convert(reflect.TypeOf(float64(0))).Float()
+		// check if the value is non-zero or if the field is in the forceSendFields list
+		if float64Val != 0 || forceSendField {
+			destField.Set(reflect.ValueOf(types.Float64Value(float64Val)))
+		} else {
+			destField.Set(reflect.ValueOf(types.Float64Null()))
+		}
+	case reflect.String:
+		var strVal string
+		if srcField.Type().Name() != "string" {
+			// This case is for Enum Types.
+			strVal = getStringFromEnum(srcField)
+		} else {
+			strVal = srcFieldValue.(string)
+		}
+		// check if the value is non-zero or if the field is in the forceSendFields list
+		if strVal != "" || forceSendField {
+			destField.Set(reflect.ValueOf(types.StringValue(strVal)))
+		} else {
+			destField.Set(reflect.ValueOf(types.StringNull()))
+		}
```

Comment:
Is the else case needed? The default value should be empty string IIUC.

Reply:
Setting `StringNull` just means that we are explicitly saying this is the case that is "unset" thus when we convert it back to gosdk struct, we know whether to add it into FSF or not